src/AppBundle/Service/CasrecVerificationService.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace AppBundle\Service;
+
+use AppBundle\Entity\CasRec;
+use Doctrine\ORM\EntityManager;
+use \Doctrine\Common\Util\Debug as doctrineDebug;
+
+class CasrecVerificationService
+{
+    /** @var EntityManager */
+    private $em;
+
+    /** @var \Doctrine\ORM\EntityRepository */
+    private $casRecRepo;
+
+    private $lastMatchedCasrecUsers;
+
+    public function __construct($em)
+    {
+        $this->em = $em;
+        $this->casRecRepo = $this->em->getRepository('AppBundle\Entity\CasRec');
+        $this->lastMatchedCasrecUsers = [];
+    }
+
+    /**
+     * CASREC checks
+     * Throw error 400 if casrec has no record matching case number,
+     * client surname, deputy surname, and postcode (if set)
+     *
+     * @param string $caseNumber
+     * @param string $clientSurname
+     * @param string $deputySurname
+     * @param string $deputyPostcode
+     * @return bool
+     */
+    public function validate($caseNumber, $clientSurname, $deputySurname, $deputyPostcode)
+    {
+        $crMatches = $this->casRecRepo->findBy( [ 'caseNumber'     => $this->normaliseCaseNumber($caseNumber)
+                                                , 'clientLastname' => $this->normaliseSurname($clientSurname)
+                                                , 'deputySurname'  => $this->normaliseSurname($deputySurname)
+                                                ]);
+
+        $this->lastMatchedCasrecUsers = $this->applyPostcodeFilter($crMatches, $deputyPostcode);
+
+        if (count($this->lastMatchedCasrecUsers) == 0) {
+            throw new \RuntimeException('User registration: no matching record in casrec', 400);
```

Comment:
im happy to leave this for now given the size of the story but an exception is something we dont expect to happen. Given the validate nature of this method simply returning false would work better here. Since the method succeeded to run and simply didnt validate